lib/diagnostic.rb

```diff
@@ -2,41 +2,42 @@
 
 # Ruby enumerable diagnostic
 class Diagnostic
+  include Enumerable
   # Question 1
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max_by(&:length)
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any? {|num| num%2 != 0}
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.all? {|num| num%2 == 0}
   end
```

Comment:
Both 3 & 4 work perfectly fine, but you are not taking advantage of the methods Ruby provides for us. Consider this solution:
```rb
  def every_even(numbers)
    numbers.all?(&:even?)
  end
```
https://ruby-doc.org/core-2.2.2/Integer.html#method-i-even-3F